src/handlers/pointer_handler.rs

```diff
@@ -0,0 +1,91 @@
+use bevy::{
+    input::{ButtonState, mouse::MouseButtonInput},
+    prelude::MouseButton,
+    window::{CursorEntered, CursorLeft, CursorMoved, Window, WindowEvent},
+};
+use smithay_client_toolkit::{
+    delegate_pointer, reexports::client::Proxy, seat::pointer::PointerHandler,
+};
+
+use crate::{smithay_windows::SmithayWindows, state::SmithayRunnerState};
+
+fn convert_to_mouse_button(button: u32) -> MouseButton {
+    match button {
+        272 => MouseButton::Left,
+        273 => MouseButton::Right,
+        274 => MouseButton::Middle,
+        277 => MouseButton::Forward,
+        278 => MouseButton::Back,
+        other => MouseButton::Other(other as u16),
+    }
+}
+
+impl PointerHandler for SmithayRunnerState {
+    fn pointer_frame(
+        &mut self,
+        _: &smithay_client_toolkit::reexports::client::Connection,
+        _: &smithay_client_toolkit::reexports::client::QueueHandle<Self>,
+        _: &smithay_client_toolkit::reexports::client::protocol::wl_pointer::WlPointer,
+        events: &[smithay_client_toolkit::seat::pointer::PointerEvent],
+    ) {
+        for event in events {
+            let smithay_windows = self.world().non_send_resource::<SmithayWindows>();
+            let window_id = event.surface.id();
+            let entity = smithay_windows.smithay_to_entity.get(&window_id);
+            if entity.is_none() {
```

Comment:
Your implementation in keyboard handler panics, but there you are handling the scenario, can we ensure consistency between handlers. Also in scenarios like these add `warn`